internal/controller/apiextensions/composite/api.go

```diff
@@ -198,27 +197,42 @@ func (f *APIRevisionFetcher) Fetch(ctx context.Context, cr resource.Composite) (
 	return AsComposition(current), nil
 }
 
-// currentRevision returns the current revision of the supplied composition. It
-// returns nil if none of the supplied revisions appear to be currentRevision.
+func (f *APIRevisionFetcher) getCompositionRevisionList(ctx context.Context, cr resource.Composite, comp *v1.Composition) (*v1alpha1.CompositionRevisionList, error) {
+	rl := &v1alpha1.CompositionRevisionList{}
+	ml := client.MatchingLabels{}
+
+	if cr.GetCompositionUpdatePolicy() != nil && *cr.GetCompositionUpdatePolicy() == xpv1.UpdateAutomatic &&
+		cr.GetCompositionRevisionSelector() != nil && len(cr.GetCompositionRevisionSelector().MatchLabels) != 0 {
+		ml = cr.GetCompositionRevisionSelector().MatchLabels
+	}
+
+	ml[v1alpha1.LabelCompositionName] = comp.GetName()
+	if err := f.ca.List(ctx, rl, ml); err != nil {
+		return nil, err
```

Comment:
Should this error be wrapped?

Reply:
It's wrapped in the Fetch method that calls getCompositionRevisionList

Reply:
wrapped the error message